src/libstd/sys/unix/fs.rs

```diff
@@ -120,21 +123,36 @@ cfg_has_statx! {{
             ) -> c_int
         }
 
-        if !HAS_STATX.load(Ordering::Relaxed) {
-            return None;
-        }
-
-        let mut buf: libc::statx = mem::zeroed();
-        let ret = cvt(statx(fd, path, flags, mask, &mut buf));
-        match ret {
-            Err(err) => match err.raw_os_error() {
-                Some(libc::ENOSYS) => {
-                    HAS_STATX.store(false, Ordering::Relaxed);
-                    return None;
+        match STATX_STATE.load(Ordering::Relaxed) {
+            // For the first time, we try to call on current working directory
+            // to check if it is available.
+            0 => {
+                let mut buf: libc::statx = mem::zeroed();
+                let err = cvt(statx(
+                    libc::AT_FDCWD,
+                    b".\0".as_ptr().cast(),
+                    0,
+                    libc::STATX_ALL,
+                    &mut buf,
+                ))
+                    .err()
+                    .and_then(|e| e.raw_os_error());
+                // `seccomp` will emit `EPERM` on denied syscall.
+                // See: https://github.com/rust-lang/rust/issues/65662
```

Comment:
Could this expand the comment inline to explain why `EPERM` is handled specially and why we attempt to stat a "known good" location, along with the pitfalls of this could still return a false negative?

Reply:
> ... along with the pitfalls of this could still return a false negative?

Will it? The manual said it never returns EPERM. So it should be the only case.

Reply:
Perhaps! I've come to not trust man pages on errors though, they basically never describe the exhaustive set of errors, only some possible ones.